pkg/util/provider/machinecontroller/machineclass.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright (c) 2017 SAP SE or an SAP affiliate company. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package controller is used to provide the core functionalities of machine-controller-manager
+package controller
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+
+	"k8s.io/klog"
+
+	"github.com/gardener/machine-controller-manager/pkg/apis/machine"
+	"github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+)
+
+// machineClassKind is used to identify the machineClassKind as AWS
+const machineClassKind = "machineClass"
+
+func (c *controller) machineDeploymentToMachineClassDelete(obj interface{}) {
+	machineDeployment, ok := obj.(*v1alpha1.MachineDeployment)
+	if machineDeployment == nil || !ok {
+		return
+	}
+	if machineDeployment.Spec.Template.Spec.Class.Kind == machineClassKind {
+		c.machineClassQueue.Add(machineDeployment.Spec.Template.Spec.Class.Name)
+	}
+}
+
+func (c *controller) machineSetToMachineClassDelete(obj interface{}) {
```

Comment:
Not really. I kept for a future use case if required. However, it can be removed here - Nice catch. Done - https://github.com/gardener/machine-controller-manager/pull/460/commits/a8db3bff253da9d0be5a0acefd4f89826f6a53d9.